src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -0,0 +1,29 @@
+package com.hibob.academy.service
+
+import com.hibob.academy.dao.Pet
+import com.hibob.academy.dao.PetCreationRequest
+import com.hibob.academy.dao.PetDao
+import jakarta.ws.rs.NotFoundException
+import org.springframework.stereotype.Component
+
+
+@Component
+class PetService(private val petDao: PetDao) {
+
+    fun createPet(petCreationRequest: PetCreationRequest): Long {
+        return petDao.createPet(petCreationRequest)
+    }
+
+    fun getPetsByType(type: String, companyId: Long): List<Pet> {
+        return petDao.getPetsByType(type, companyId)
+    }
+
+    fun getPetById(id: Long, companyId: Long): Pet? {
```

Comment:
Do we want to return `Pet?` or `Pet` ?
Why not handle the `null` here?

Reply:
Yes its better i changed it